### PR TITLE
Attachments: Illumina Paired End detection

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates the `Attachments::CreateService` to identify Illumina Paired End attachments when more than 1 `fastq` file is successfully attached to the attachable.

When both the `forward` and `reverse` file is detected for `fastq` files that match the [Illumina Fastq naming convention](https://help.basespace.illumina.com/files-used-by-basespace/fastq-files#naming) add the following metadata:

Forward:
```json
{
  "type": "illumina_pe",
  "direction": "forward",
  "associated_attachment_id": 2
}
```

Reverse:
```json
{
  "type": "illumina_pe",
  "direction": "reverse",
  "associated_attachment_id": 1
}
```

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as your user of choice, and create a personal project
2. Create a Sample in the project you just created
3. Navigate to the Sample
4. Click `Upload Files`
5. Select `test/fixtures/files/TestSample_S1_L001_R1_001.fastq` and `test/fixtures/files/TestSample_S1_L001_R2_001.fastq`
6. Click `Upload`
7. In your Rails console, execute `Attachments.last` see that it has the following metadata fields `type`, `direction`, and `associated_attachment_id`
8. In your Rails console, execute `Attachments.last.associated_attachment` see that it has the following metadata fields `type`, `direction`, and `associated_attachment_id`

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/services/attachments/create_service.rb [hunk 1]
```diff
@@ -22,7 +22,51 @@ def execute
...
+      persisted_fastq_attachments = @attachments.select { |attachment| attachment.persisted? && attachment.fastq? }
+
+      identify_illumina_pe(persisted_fastq_attachments) if persisted_fastq_attachments.count > 1
+
       @attachments
     end
+
+    private
+
+    def identify_illumina_pe(attachments) # rubocop:disable Metrics/MethodLength,Metrics/CyclomaticComplexity,Metrics/AbcSize
```
Comment (deepsidhu85) [inline app/services/attachments/create_service.rb:35]: Can we update this method name so it's clear. The `pe` part of the method anyways.

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good to me and is working great! Tested it out with a bunch of different fastq and fastq.gz files. Just have the one comment below for the method name

File: app/services/attachments/create_service.rb (hunk 1 above)
Comment (ericenns) [reply app/services/attachments/create_service.rb]: Addressed in b04242c

**Review by deepsidhu85 — APPROVED** [review]
This looks good to me. Thanks for making the change and adding tests!

**Pull request merged**